components/infobox/wikis/apexlegends/infobox_weapon_custom.lua

```diff
@@ -0,0 +1,176 @@
+---
+-- @Liquipedia
+-- wiki=apexlegends
+-- page=Module:Infobox/Weapon/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Weapon = require('Module:Infobox/Weapon')
+local Class = require('Module:Class')
+local String = require('Module:StringUtils')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local CustomWeapon = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _weapon
+local _args
+
+local MAGAZINE_INFO = {
+	{},
+	{{text = '[[with Common Ext. Mag|<span class="white-text">EXT</span>]]', bgClass = 'gray-theme-dark-bg'}},
+	{{text = '[[with Rare Ext. Mag|<span class="white-text">EXT</span>]]', bgClass = 'sapphire-a2'}},
+	{
+		{text = '[[with Epic Ext. Mag|<span class="white-text">EXT</span>]]', bgClass = 'vivid-violet-theme-dark-bg'},
+		{text = '[[with Legendary Ext. Mag|<span class="black-text">EXT</span>]]', bgClass = 'bright-sun-0'}
+	},
+}
+local DAMAGE_INFO = {
+	{},
+	{{text = 'Head', bgClass = 'gray-theme-dark-bg', textBgClass = 'white-text'}},
+	{{text = 'Leg', bgClass = 'gray-theme-light-bg', textBgClass = 'black-text'}},
+}
+local NON_BREAKING_SPACE = '&nbsp;'
+
+function CustomWeapon.run(frame)
+	local weapon = Weapon(frame)
+	_weapon = weapon
+	_args = _weapon.args
+	weapon.addToLpdb = CustomWeapon.addToLpdb
+	weapon.createWidgetInjector = CustomWeapon.createWidgetInjector
+	return weapon:createInfobox(frame)
+end
+
+function CustomWeapon:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	local args = _args
+	if String.isNotEmpty(args.basedamage) and String.isEmpty(args.damage) then
+		local baseDamages = {}
+		for index, baseDamage in ipairs(_weapon:getAllArgsForBase(args, 'basedamage')) do
+			table.insert(baseDamages, tostring(CustomWeapon:_createContextualNoWrappingSpan(baseDamage, index, DAMAGE_INFO)))
```

Comment:
Is this `tostring` actually required here?

Reply:
imo the `tostring` should be moved to line 167 and removed around the calls of the function

Reply:
The `tostring` was just to allow it to be inserted into the widget cell on line 61 with `table.concat()` method. Would there  be a better way to insert the data to the field after constructing it?

Reply:
Moved the `tostring()` call to the Wrapping function as per @hjpalpha suggestion